docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,103 @@
+# Architecture
+
+An overview of the architecture for various components within the Toolkit.
+
+## Webviews
```

Comment:
I would clarify that this isn't a requirement for all webviews. e.g. the Step Functions webview which brings its own JS payload.

Reply:
True, though I'd say that the existence of the webview is not desirable (it can't be manipulated through shared code)